Actions/ProcessALCodeAnalysisLogs/ProcessALCodeAnalysisLogs.ps1

```diff
@@ -29,36 +96,57 @@ function GenerateSARIFJson {
     )
 
     foreach ($issue in $errorLogContent.issues) {
-        # Add rule if not already added
-        if (-not ($sarif.runs[0].tool.driver.rules | Where-Object { $_.id -eq $issue.ruleId })) {
-            $sarif.runs[0].tool.driver.rules += @{
-                id = $issue.ruleId
-                shortDescription = @{ text = $issue.fullMessage }
-                fullDescription = @{ text = $issue.fullMessage }
-                helpUri = $issue.properties.helpLink
-                properties = @{
-                    category = $issue.properties.category
-                    severity = $issue.properties.severity
-                }
-            }
+        # Skip issues without locations as GitHub expects at least one location
+        if (($issue.PSObject.Properties.Name -notcontains "locations" ) -or ($issue.locations.Count -eq 0)) {
+            continue
+        }
+
+        $newResult = $null
+        $relativePath = Get-FileFromAbsolutePath -AbsolutePath $issue.locations[0].analysisTarget[0].uri
+        $message = Get-IssueMessage -issue $issue
+
+        # Skip issues if we cannot find a message
+        if ($null -eq $message) {
+            OutputDebug -message "Could not extract message from issue: $($issue | ConvertTo-Json -Depth 10 -Compress)"
+            continue
         }
 
-        # Convert absolute path to relative path from repository root
-        $absolutePath = $issue.locations[0].analysisTarget[0].uri
-        $workspacePath = $ENV:GITHUB_WORKSPACE
-        $relativePath = $absolutePath.Replace($workspacePath, '').TrimStart('\').Replace('\', '/')
+        # Skip issues if we cannot find the file in the workspace
+        if ($null -eq $relativePath) {
+            OutputDebug -message "Could not find file for issue: $($issue | ConvertTo-Json -Depth 10 -Compress)"
+            continue
+        }
 
-        # Add result
-        if (-not ($sarif.runs[0].results | Where-Object {
+        # Check if result already exists in the sarif object
+        $existingResults = $sarif.runs[0].results | Where-Object {
             $_.ruleId -eq $issue.ruleId -and
-            $_.message.text -eq $issue.shortMessage -and
-            $_.locations[0].physicalLocation.artifactLocation.uri -eq $relativePath -and
-            ($_.locations[0].physicalLocation.region | ConvertTo-Json) -eq ($issue.locations[0].analysisTarget[0].region | ConvertTo-Json) -and
-            $_.level -eq ($issue.properties.severity).ToLower()
-        })) {
-            $sarif.runs[0].results += @{
+            $_.message.text -eq $message -and
+            $_.level -eq ($issue.properties.severity).ToLower() -and
+            ($_.locations[0].physicalLocation.artifactLocation.uri -eq $relativePath) -and
+            ($_.locations[0].physicalLocation.region | ConvertTo-Json) -eq ($issue.locations[0].analysisTarget[0].region | ConvertTo-Json)
+        }
+
+        # Add result if it does not already exist
+        if (-not $existingResults)
```

Comment:
Minor: I think the following fits better in this function:

```
if ($existingResults) {
    # Skip if existing
    continue
}
```

That way you can remove the final `if ($null -ne $newResult)` check as well.